lib/spack/spack/cmd/buildcache.py

```diff
@@ -383,10 +436,30 @@ def _concrete_spec_from_args(args):
 
 def create_fn(args):
     """create a binary package and push it to a mirror"""
-    push_url = _mirror_url_from_args(args)
-    matches = _matching_specs(args)
+    if args.mirror_flag:
+        mirror = args.mirror_flag
+    elif not args.mirror:
+        raise ValueError("No mirror provided")
+    else:
+        mirror = arguments.mirror_name_or_url(args.mirror)
+
+    if args.mirror_flag:
+        tty.warn(
+            "Using flags to specify mirrors is deprecated and will be removed in "
+            "Spack 0.21, use positional arguments instead."
+        )
+
+    # TODO: remove this in 0.21. If we have mirror_flag, the first
+    # spec is in the positional mirror arg due to argparse limitations.
+    specs = args.specs
+    if args.mirror_flag and args.mirror:
+        specs.insert(0, args.mirror)
```

Comment:
I tried this locally with just the mirror arg and it doesn't seem to work:

```console
> spack mirror create ./cache zlib
==> Concretizing input specs
==> Error: unexpected tokens in the spec string
./cache zlib
^
```

```console
> spack mirror create cache zlib
==> Concretizing input specs
==> Error: Package 'cache' not found.
You may need to run 'spack clean -m'.
```

```console
> spack mirror create cache
==> Concretizing input specs
==> Error: Package 'cache' not found.
You may need to run 'spack clean -m'.
```

I believe that's because it's trying to concretize the specs in the argparse action... but the first spec ends up being the directory name.  Right?  Or am I doing something wrong?

Reply:
Heh, I did not touch `spack mirror create`, only `spack buildcache create`. I suppose it should have the same signature as `spack buildcache create` then.

Reply:
I guess I shouldn’t review at 1am 🤦🏻

Reply:
And yeah they should probably be symmetric!

Reply:
Unfortunately a positional argument is breaking :sob: 

`spack mirror create [specs...]` without `--directory` mirrors sources into a default directory.

I don't want this PR to be breaking, so I'm afraid `spack mirror create` will be the exception :(